### PR TITLE
:bug: Exclude virtual tags from writing/updating

### PR DESCRIPTION
**Pull request by mansam**
* Add `virtual` field to TagRef
* Ignore tag refs with `virtual` field set true when adding, updating, or replacing tags.
* Fix bug preventing archetype tags being inherited properly
* go fmt

### CODE REVIEW

File: api/application.go [hunk 1]
```diff
@@ -140,8 +147,8 @@ func (h ApplicationHandler) Get(ctx *gin.Context) {
 	r := Application{}
 	r.With(m, tags)
 	r.WithArchetypes(archetypes)
-	r.WithSourcedTags(archetypeTags, "archetype")
-	r.WithSourcedTags(resolver.AssessmentTags(), "assessment")
+	r.WithVirtualTags(archetypeTags, "archetype")
+	r.WithVirtualTags(resolver.AssessmentTags(), "assessment")
```
Comment (jortel) [inline api/application.go:151]: Did you mean to use new the _const_ here?

File: migration/v9/model/archetype.go [hunk 2]
```diff
@@ -7,8 +7,8 @@ type Archetype struct {
 	Comments          string
 	Review            *Review            `gorm:"constraint:OnDelete:CASCADE"`
 	Assessments       []Assessment       `gorm:"constraint:OnDelete:CASCADE"`
-	CriteriaTags      []Tag              `gorm:"many2many:ArchetypeCriteriaTags"`
-	Tags              []Tag              `gorm:"many2many:ArchetypeTags"`
+	CriteriaTags      []Tag              `gorm:"many2many:ArchetypeCriteriaTags;constraint:OnDelete:CASCADE"`
+	Tags              []Tag              `gorm:"many2many:ArchetypeTags;constraint:OnDelete:CASCADE"`
```
Comment (jortel) [inline migration/v9/model/archetype.go:11]: I assume this fixes archetype delete.

**Review by jortel — APPROVED** [review]
LGTM
One comment re: usage of _const_.

File: api/application.go (hunk 1 above)
Comment (mansam) [reply api/application.go:151]: I did, thanks for pointing that out.

File: migration/v9/model/archetype.go (hunk 2 above)
Comment (mansam) [reply migration/v9/model/archetype.go:11]: This already exists on main so I'm not sure why it's showing up separately. I'll rebase my PR and clean it up.

**Pull request merged**